blog/articles/set-up-a-local-knative-environment-up-in-no-time-flat.md

```diff
@@ -0,0 +1,310 @@
+---
+Title: 'How to set up a local Knative environment up in no time flat'
+Author: Leon Stigter
+Author handle: https://twitter.com/retgits
+Date: ''
+Description: A how-to guide to deploy Knative, Kourier, and your first app on top of a Kubernetes cluster.
+Folder with media files: 'N/A'
+Blog URL: ''
+Labels: Articles
+Reviewers: ''
+
+---
+| Reviewer  | Date | Approval |
+| ------------- | ------------- | ------------- |
+| @retgits  | 2020-06-03  |:+1:|
+| <!-- Your Github handle here --> |   |  |
+
+Knative builds on top of Kubernetes. For developers, it abstracts away complexity and enables them to focus on delivering value to their business. The complex, and sometimes boring, parts of building apps to run on Kubernetes are managed by Knative. Setting up a Kubernetes cluster and starting with Knative on your local machine can still be complex. In this post, let’s focus on setting up a lightweight environment to help you modern apps faster.
+
+## KinD
+If you want to create a Kubernetes cluster on your local machine, you have a bunch of options. Because you’re running containers in your Kubernetes cluster anyway, let’s also use containers for the actual cluster itself. Kubernetes IN Docker, or `kind` for short, enables developers to spin up a Kubernetes cluster where each “node” is a container.
+
+You can install `kind` on your machine, by running the commands below:
+
+```bash
+curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.8.1/kind-$(uname)-amd64
+chmod +x ./kind
+mv ./kind /some-dir-in-your-PATH/kind
+```
+
+The next step is to create a Kubernetes cluster using `kind`. While you can certainly use `kubectl port-forward` to get traffic into the cluster, a nicer way is to expose the ports the ingress gateway will listen on to the host. To do that, you can pass in a file with cluster configuration parameters:
+
+```bash
+cat > clusterconfig.yaml <<EOF
+kind: Cluster
+apiVersion: kind.sigs.k8s.io/v1alpha3
+nodes:
+- role: control-plane
+  extraPortMappings:
+	## expose port 31380 of the node to port 80 on the host
+  - containerPort: 31080
```

Comment:
maybe explain more here explicity the little magic here.
Tell the user that the value `31080` is arbitraty that will be use later to configure a service with NodePort using the same value

Reply:
I've added some additional explanation underneath the code block.